source/Superluminal-Model-Tests/AddMultiPartCommandTest.class.st

```diff
@@ -0,0 +1,41 @@
+Class {
+	#name : #AddMultiPartCommandTest,
+	#superclass : #AbstractEnvironmentTestCase,
+	#category : #'Superluminal-Model-Tests-Commands'
+}
+
+{ #category : #tests }
+AddMultiPartCommandTest >> testApplyOn [
+
+	| httpClient command |
+
+	httpClient := ZnClient new.
+	command := AddMultiPartCommand named: 'name' pairedTo: 'fran'.
+	command applyOn: httpClient.
+
+	self
+		assert: httpClient request contentType main equals: 'multipart';
+		assert: httpClient request contentType sub equals: 'form-data'.
+	self assert: ( httpClient request entity partNamed: 'name' ) fieldValue equals: 'fran'
+]
+
+{ #category : #tests }
+AddMultiPartCommandTest >> testApplyOnWhenAttachingFiles [
+
+	|  filename httpClient command |
+
+	httpClient := ZnClient new.
+	ZnFileSystemUtils deleteIfExists: 'testApplyOnWhenAttachingFiles.txt'.
```

Comment:
I suggest to perform an extract method a method named `#onFileNamed:withContents:assert:`

```smalltalk
self onFileNamed: 'testApplyOnWhenAttachingFiles.txt' withContents: '0123456789' assert: [:fileName :contents | 
  command := AddMultiPartCommand named: 'file' attaching: fileName  asFileReference.
  command applyOn: httpClient.

 	self
 		assert: httpClient request contentType main equals: 'multipart';
 		assert: httpClient request contentType sub equals: 'form-data'.

 	self assert: ( httpClient request entity partNamed: 'file' ) fieldValue equals: contents
 ]
```   

Do the same refactoring to the other methods.